### ISSUE TITLE
`var`: assert when reducing over multiple dims including 1-len dims.

### ISSUE THREAD

**Issue by nikitaved** [NVIDIA/Fuser#1721]
The following snippet:

```python                                                                                                                                                                                     
import torch                                                                                                                                                                                  
from nvfuser import FusionDefinition, DataType                                                                                                                                                
                                                                                                                                                                                              
def nvfuser_fusion_id4(fd : FusionDefinition) -> None :                                                                                                                                       
    T0 = fd.define_tensor(shape=[-1, -1, -1, 1], contiguity=[True, True, True, None], dtype=DataType.Float, is_cpu=False, stride_order=[3, 2, 1, 0])                                          
    S1 = fd.define_scalar(None, dtype=DataType.Int)                                                                                                                                           
    S2 = fd.define_scalar(None, dtype=DataType.Int)                                                                                                                                           
    S3 = fd.define_scalar(None, dtype=DataType.Int)                                                                                                                                           
    S4 = fd.define_scalar(None, dtype=DataType.Bool)                                                                                                                                          
    T5 = fd.ops.var(T0, axes=[1, 3], correction=1, keepdim=False)                                                                                                                             
    fd.add_output(T5)                                                                                                                                                                         
                                                                                                                                                                                              
with FusionDefinition() as fd:                                                                                                                                                                
    nvfuser_fusion_id4(fd)                                                                                                                                                                    
                                                                                                                                                                                              
inputs = [                                                                                                                                                                                    
    torch.randn((280,), dtype=torch.float32, device='cuda:0').as_strided((8, 7, 5, 1), (35, 5, 1, 1)),                                                                                        
    1,                                                                                                                                                                                        
    1,                                                                                                                                                                                        
    3,                                                                                                                                                                                        
    False,                                                                                                                                                                                    
]                                                                                                                                                                                             
fd.execute(inputs)                                                                                                                                                                            
```                          

fails with

```                                                                                                                  
RuntimeError: known_size == this_size INTERNAL ASSERT FAILED at "/workspace/Fuser/csrc/expr_evaluator.cpp":293, please report a bug with repro script to NVFuser at https://github.com/NVIDIA/
Fuser/issues. Conflicting sizes: 8, 7                                                                                                                                                         
Exception raised from propagateBoundValuesThroughExactMaps at /workspace/Fuser/csrc/expr_evaluator.cpp:293 (most recent call first):

```

**Comment by jacobhinkle** [comment]
Will be fixed by #1717. Sorry about the breakage!

**Comment by nikitaved** [comment]
@jacobhinkle , I will also update the logic with squeeze on our side. I agree that it is just easier to be consistent with PyTorch.
